### ISSUE TITLE
FileSystemNotFoundException on standalone Spring Boot app

### ISSUE THREAD

**Issue by ivanbportugal** [rulebook-rules/rulebook#91]
I like where this library is going so I am giving it a shot. The main problem I've encountered that I don't seem to find a workaround for is this `FileSystemNotFoundException` that occurs when I run the RuleBook standalone (I am using the annotated classes not the Java DSL). Everything works fine in IntelliJ but if I try to just `java -jar target/long-jar-name.jar` from the command line it fails. I am running this in a standard Spring Boot app. 

It appears that the problem could be how the Paths are created using NIO: https://stackoverflow.com/questions/22605666/java-access-files-in-jar-causes-java-nio-file-filesystemnotfoundexception

I am using version 0.8 pulled from Maven Central. Here is what the exception looks like:
```
2017-09-12 23:18:22 [http-nio-8080-exec-3] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.nio.file.FileSystemNotFoundException] with root cause
java.nio.file.FileSystemNotFoundException: null
        at com.sun.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:171)
        at com.sun.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:157)
        at java.nio.file.Paths.get(Paths.java:143)
        at com.deliveredtechnologies.rulebook.model.runner.RuleBookRunner.findRuleClassesInPackage(RuleBookRunner.java:120)
        at com.deliveredtechnologies.rulebook.model.runner.RuleBookRunner.run(RuleBookRunner.java:68)
```

**Comment by Clayton7510** [comment]
Thanks for raising this issue. There is a fix in place in the 0.8.1-SNAPSHOT release now that was tested with a Spring Boot packaged Jar. I'm planning on releasing v0.8.1 sometime this weekend.

**Comment by eitansela** [comment]
From where can I take this fix? It is not in maven repository...

**Comment by Clayton7510** [comment]
It's in the sonatype snapshot maven repo: "https://oss.sonatype.org/content/repositories/snapshots"

**Comment by Clayton7510** [comment]
Here are some examples of how to add the snapshot repo to your project: https://github.com/rulebook-rules/rulebook/wiki/Download-and-Install

**Comment by eitansela** [comment]
Succeeded to use sonatype snapshot maven repo.
However, with 0.8.1-SNAPSHOT , when trying to run standalone executable Spring-Boot jar,I get the following exception:

`14-09-2017 07:12:26.350 [XNIO-2 task-1] ERROR c.d.r.model.runner.RuleBookRunner - Unable to find rule classes in package 'com.my.example'
java.nio.file.InvalidPathException: com.my.example: 'com.my.example' is not a valid path
	at com.deliveredtechnologies.rulebook.model.runner.RuleBookRunner.findRuleClassesInPackage(RuleBookRunner.java:148)
	at com.deliveredtechnologies.rulebook.model.runner.RuleBookRunner.run(RuleBookRunner.java:73)
	at com.weissbeerger.recommendation.engine.controller.PortfolioPageRuleController.getRecommenations(PortfolioPageRuleController.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:133)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:97)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)`

In eclipse IDE it works flawlessly.

**Comment by Clayton7510** [comment]
Are you using a Spring Boot war deployed on Wildfly? Or a standalone Spring Boot jar w/Undertow? Or, could you attach your source?

I tested this with the Spring Boot example project under rulebook-examples. 

But I'll validate the other scenarios listed above and add examples to validate this evening. And of course, if you can provide your specific example, I can ensure it works with that too.

**Comment by eitansela** [comment]
I use standalone Spring Boot jar w/Undertow

**Comment by Clayton7510** [comment]
Interesting, that's what I also use here in the _helloworld-spring-web_ app in the _Issue90_PojoResults_ branch. And I didn't run into that issue; it worked as expected. The project can be seen here: https://github.com/rulebook-rules/rulebook/tree/Issue90_PojoResults/rulebook-examples

Would you mind attaching your example or providing a link so I can see what the issue is? Judging by your error, it appears as if RuleBook cannot find the package that contains the POJO rules to scan. However, since you said it is working in your IDE, I have to wonder what's specifically going on there.

**Comment by eitansela** [comment]
After hours of digging and comparing my code to your example Spring Boot application, I finally found the problem!

It is related to the following in build.gradle:
`springBoot {
    executable = true
}`

This is done as part of Spring Boot convention to enable Spring Boot applications run as Linux in systemd services, as detailed here:
https://docs.spring.io/spring-boot/docs/current/reference/html/deployment-install.html

Once I removed that part, it worked!
This is actually a good news, since we will not use Spring Boot applications run as Linux in systemd, but as part of a Docker image. So it is redundant and can be deleted. 

Thanks for all your help! Especially for the Spring Boot example application you posted.
Keep up the good work :-)
